native/acter/src/api/account.rs

```diff
@@ -72,6 +72,11 @@ impl Account {
     pub async fn upload_avatar(&self, uri: String) -> Result<OwnedMxcUri> {
         let account = self.account.clone();
         let path = PathBuf::from(uri);
+        let guess = mime_guess::from_path(path.clone());
+        let content_type = guess.first().context("MIME type should be given")?;
```

Comment:
This would be the user facing error, probably want to do something a bit more user friendly:

```suggestion
        let content_type = guess.first().context("Could not detect mime type. Are you sure this is an image?")?;
```